lib/modules/apostrophe-caches/index.js

```diff
@@ -242,5 +243,26 @@ module.exports = {
         }
       );
     };
+
+    self.removeBadIndexMigration = function() {
+      // Cache module wakes before migrations module so we have to wait
+      // for apostrophe:modulesReady before registering the migration. Most modules
+      // can register one directly from afterConstruct
+      self.on('apostrophe:modulesReady', 'addRemoveBadIndexMigration', function() {
+        self.apos.migrations.add(self.__meta.name + '.removeBadIndex', async function() {
+          const indexes = await self.cacheCollection.indexes();
+          const culprit = indexes.find(index => index.key &&
+            (Object.keys(index.key).length === 2) &&
+            index.key.key &&
+            index.key.cache &&
```

Comment:
This property never existed. I should have written `name`. Since it was always null/undefined I effectively had a unique index on `key` alone, breaking support for the key in two differently named caches, whose entries differ on `name`, not `cache`.